### ISSUE TITLE
hs.load() fails on a filename with closed square brackets

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1325]
A signal with filename `"signal[200-300]eV.hdf5"` is successfully saved with hyperspy, but it fails to load.  I think the square brackets are not being escaped properly.

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-40-e948282fe202> in <module>()
----> 1 b = hs.load(path + 'signal[200-300]eV.hdf5')

/Users/thomas/Dropbox/0_Git/GitHub/hyperspy_aar/hyperspy/io.py in load(filenames, signal_type, stack, stack_axis, new_axis_name, mmap, mmap_dir, **kwds)
    146                                if os.path.isfile(f)])
    147         if not filenames:
--> 148             raise ValueError('No file name matches this pattern')
    149     elif not isinstance(filenames, (list, tuple)):
    150         raise ValueError(

ValueError: No file name matches this pattern

**Comment by vidartf** [comment]
Can reproduce, I think these are the offending code lines:

``` python
> c:\miniconda3\envs\hspy\lib\site-packages\hyperspy\io.py(145)load()
    143
    144     if isinstance(filenames, str):
--> 145         filenames = natsorted([f for f in glob.glob(filenames)
    146                                if os.path.isfile(f)])
    147         if not filenames:
```

`glob.glob()` here returns an empty list, because it thinks the square brackets signify a character filter: https://docs.python.org/2/library/glob.html

**Comment by francisco-dlp** [comment]
Good catch! This bug must be as old as HyperSpy!

**Comment by thomasaarholt** [comment]
Now I feel real smart. I just rediscovered this bug, and searched the hyperspy repo for any issues on it. Guess who found it last year....

**Comment by vidartf** [comment]
This year you're in a much better position to write a fix though 😄

**Comment by thomasaarholt** [comment]
Fixed in #1757.

**Comment by francisco-dlp** [comment]
This may actually not need other fixing than documentations because using ``'`` instead of ``"`` works.
